unit_testing/DC/DC_tests.cpp

```diff
@@ -0,0 +1,197 @@
+        ** MEMBERS
+        *******************************************************************************************/
+        
+        ADC_HandleTypeDef hadc;
+        WWDG_HandleTypeDef hwwdg;
+        const char * bootMsg = "Boot Unit Test";
+        int tickCounter = 0;
+};
+
+/***************************************************************************************************
+** TESTS
+***************************************************************************************************/
+
+TEST_F(DCBoard, CorrectBoardParams)
+{
+    DCBoardInit(&hadc, &hwwdg);
+    setDcCurrentBuffer();
+
+    /* Basic test, was everything OK?  */
+    EXPECT_FALSE(bsGetStatus() && BS_VERSION_ERROR_Msk);
+
+    /* This should force a print on the USB bus */
+    DCBoardLoop(bootMsg);
+    HAL_ADC_ConvHalfCpltCallback(&hadc);
+    DCBoardLoop(bootMsg);
+
+    /* Check the printout is correct */
+    EXPECT_READ_USB(Contains("0.00, 0.00, 0.00, 0.00, 0.00, 0.00, 0x0"));
+}
+
+TEST_F(DCBoard, printSerial) 
+{
+    DCBoardInit(&hadc, &hwwdg);
+    /* Note: usb RX buffer is flushed during the first loop, so a single loop must be done before
+    ** printing anything */
+    DCBoardLoop(bootMsg);
+    writeDcMessage("Serial\n");
+    
+    EXPECT_FLUSH_USB(ElementsAre(
+        "\r", 
+        "Boot Unit Test\r", 
+        "Serial Number: 000\r", 
+        "Product Type: DC Board\r", 
+        "Sub Product Type: 0\r", 
+        "MCU Family: Unknown 0x  0 Rev 0\r", 
+        "Software Version: 0\r", 
+        "Compile Date: 0\r", 
+        "Git SHA: 0\r", 
+        "PCB Version: 3.1\r"
+    ));
+}
+
+TEST_F(DCBoard, printStatus) 
+{
+    DCBoardInit(&hadc, &hwwdg);
+    /* Note: usb RX buffer is flushed during the first loop, so a single loop must be done before
+    ** printing anything */
+    DCBoardLoop(bootMsg);
+    writeDcMessage("Status\n");
+    
+
+    EXPECT_FLUSH_USB(ElementsAre(
+        "\r", 
+        "Boot Unit Test\r", 
+        "Board Status:\r", 
+        "The board is operating normally.\r", 
+        "Port 0: On: 0, PWM percent: 0\r", 
+        "Port 1: On: 0, PWM percent: 0\r", 
+        "Port 2: On: 0, PWM percent: 0\r", 
+        "Port 3: On: 0, PWM percent: 0\r", 
+        "Port 4: On: 0, PWM percent: 0\r", 
+        "Port 5: On: 0, PWM percent: 0\r", 
+        "\r", 
+        "End of board status. \r"
+    ));
+    
+    writeDcMessage("p2 on\n");
+    writeDcMessage("p4 on\n");
+    writeDcMessage("p6 on\n");
+    writeDcMessage("Status\n");
+    
+    EXPECT_FLUSH_USB(ElementsAre( 
+        "\r", 
+        "Board Status:\r", 
+        "The board is operating normally.\r", 
+        "Port 0: On: 0, PWM percent: 0\r",
+        "Port 1: On: 1, PWM percent: 999\r",
+        "Port 2: On: 0, PWM percent: 0\r",
+        "Port 3: On: 1, PWM percent: 999\r",
+        "Port 4: On: 0, PWM percent: 0\r",
+        "Port 5: On: 1, PWM percent: 999\r",
+        "\r", 
+        "End of board status. \r"
```

Comment:
Let's have part of the test check that the error bit is set if the 24V supply is missing.